test/index.js

```diff
@@ -712,3 +712,77 @@ test('\nmin satisfying', function(t) {
   });
   t.end();
 });
+
+test('\nintersect comparators', function(t) {
+  [
+    // One is a Version
+    ['1.3.0', '>=1.3.0', true],
+    ['1.3.0', '>1.3.0', false],
+    ['>=1.3.0', '1.3.0', true],
+    ['>1.3.0', '1.3.0', false],
+    // Same direction increasing
+    ['>1.3.0', '>1.2.0', true],
+    ['>1.2.0', '>1.3.0', true],
+    ['>=1.2.0', '>1.3.0', true],
+    ['>1.2.0', '>=1.3.0', true],
+    // Same direction decreasing
+    ['<1.3.0', '<1.2.0', true],
+    ['<1.2.0', '<1.3.0', true],
+    ['<=1.2.0', '<1.3.0', true],
+    ['<1.2.0', '<=1.3.0', true],
+    // Different directions, same semver and inclusive operator
+    ['>=1.3.0', '<=1.3.0', true],
+    ['>=1.3.0', '>=1.3.0', true],
+    ['<=1.3.0', '<=1.3.0', true],
+    ['>1.3.0', '<=1.3.0', false],
+    ['>=1.3.0', '<1.3.0', false],
+    // Opposite matching directions
```

Comment:
I mean, line 752 asks if comparator1 intersects comparator2; could you also add an assertion after that asking if comparator2 intersects comparator1?